sys.modules[None] gets pseudo module when pytest skip condition is string

In pypa/setuptools#860, the test_sandbox tests started failing when another commit was made that invoked a pytest skip with a string expression, because that causes `sys.modules` to get a None key in that dict. This behavior can be minimally replicated thus:

```
$ cat > test_me.py
import sys
import pytest

@pytest.mark.skipif('os.environ.get("foo")')
def test_this():
    assert None not in sys.modules        
$ python -m rwt pytest -- -m pytest
Collecting pytest
  Using cached pytest-3.0.4-py2.py3-none-any.whl
Collecting py>=1.4.29 (from pytest)
  Using cached py-1.4.31-py2.py3-none-any.whl
Installing collected packages: py, pytest
Successfully installed py-1.4.31 pytest-3.0.4
====================================== test session starts =======================================
platform darwin -- Python 3.6.0b4, pytest-3.0.4, py-1.4.31, pluggy-0.4.0
rootdir: /Users/jaraco/issue-nnn, inifile: 
collected 1 items 

test_me.py F

============================================ FAILURES ============================================
___________________________________________ test_this ____________________________________________

    @pytest.mark.skipif('os.environ.get("foo")')
    def test_this():
>       assert None not in sys.modules
E       assert None not in {'builtins': <module 'builtins' (built-in)>, 'sys': <module 'sys' (built-in)>, '_frozen_importlib': <module 'importlib._bootstrap' (frozen)>, '_imp': <module '_imp' (built-in)>, ...}
E        +  where {'builtins': <module 'builtins' (built-in)>, 'sys': <module 'sys' (built-in)>, '_frozen_importlib': <module 'importlib._bootstrap' (frozen)>, '_imp': <module '_imp' (built-in)>, ...} = sys.modules

test_me.py:6: AssertionError
==================================== 1 failed in 0.04 seconds ====================================
$ 
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -18,7 +18,10 @@
 * Fix error message using ``approx`` with complex numbers (`#2082`_).
   Thanks `@adler-j`_ for the report and `@nicoddemus`_ for the PR.
 
-*
+* Remove internal code meant to support earlier Python 3 versions that produced the side effect
+  of leaving ``None`` in ``sys.modules`` when expressions were evaluated by pytest (for example passing a condition
+  as a string to ``pytest.mark.skipif``)(`#2103`_).
+  Thanks `@jaraco`_ for the report and `@nicoddemus`_ for the PR.
 
 * Cope gracefully with a .pyc file with no matching .py file (`#2038`_). Thanks
   `@nedbat`_.
@@ -41,6 +44,7 @@
 .. _#2038: https://github.com/pytest-dev/pytest/issues/2038
 .. _#2078: https://github.com/pytest-dev/pytest/issues/2078
 .. _#2082: https://github.com/pytest-dev/pytest/issues/2082
+.. _#2103: https://github.com/pytest-dev/pytest/issues/2103
 
 
 3.0.4

--- a/_pytest/_code/source.py
+++ b/_pytest/_code/source.py
@@ -192,14 +192,6 @@ class Source(object):
             if flag & _AST_FLAG:
                 return co
             lines = [(x + "\n") for x in self.lines]
-            if sys.version_info[0] >= 3:
-                # XXX py3's inspect.getsourcefile() checks for a module
-                # and a pep302 __loader__ ... we don't have a module
-                # at code compile-time so we need to fake it here
-                m = ModuleType("_pycodecompile_pseudo_module")
-                py.std.inspect.modulesbyfile[filename] = None
-                py.std.sys.modules[None] = m
-                m.__loader__ = 1
             py.std.linecache.cache[filename] = (1, None, lines, filename)
             return co
 

--- a/_pytest/_code/source.py
+++ b/_pytest/_code/source.py
@@ -4,7 +4,6 @@ from bisect import bisect_right
 import sys
 import inspect, tokenize
 import py
-from types import ModuleType
 cpy_compile = compile
 
 try:

